sympy/functions/special/delta_functions.py

```diff
@@ -465,10 +494,18 @@ def _eval_rewrite_as_Piecewise(self, arg):
            Piecewise((0, x**2 - 1 < 0), (Heaviside(0), Eq(x**2 - 1, 0)), (1, x**2 - 1 > 0))
 
         """
-        return Piecewise((0, arg < 0), (Heaviside(0), Eq(arg, 0)), (1, arg > 0))
-
-    def _eval_rewrite_as_sign(self, arg):
+        if H0 == None:
+            return Piecewise((0, arg < 0), (Heaviside(0), Eq(arg, 0)), (1, arg > 0))
+        if H0 == 0:
+            return Piecewise((0, arg <= 0), (1, arg > 0))
+        if H0 == 1:
```

Comment:
Any advantage/disadvantage of writing `if H0 is S.Zero` and `if H0 is S.One` here?  (I'm not sure).